io/EptReader.cpp

```diff
@@ -241,23 +265,45 @@ void EptReader::addDimensions(PointLayoutPtr layout)
     const Json::Value& schema(m_info->schema());
     m_remoteLayout.reset(new FixedPointLayout());
 
-    // Register XYZ directly since they may appear as scaled int32 values in the
-    // schema.  Either way we want doubles.
-    layout->registerDim(Dimension::Id::X);
-    layout->registerDim(Dimension::Id::Y);
-    layout->registerDim(Dimension::Id::Z);
-
     for (Json::ArrayIndex i(0); i < schema.size(); ++i)
     {
         const Json::Value& dim(schema[i]);
         const std::string name(dim["name"].asString());
-        log()->get(LogLevel::Debug) << "Registering dim " << name << std::endl;
-        const Dimension::Type type(Dimension::type(dim["type"].asString()));
+
+        // If the dimension has a scale, make sure we register it as a double
+        // rather than its serialized type.
+        const Dimension::Type type = getType(dim);
+
+        log()->get(LogLevel::Debug) << "Registering dim " << name << ": " <<
+            Dimension::interpretationName(type) << std::endl;
+
         layout->registerOrAssignDim(name, type);
         m_remoteLayout->registerOrAssignDim(name, type);
     }
 
     m_remoteLayout->finalize();
+
+    using D = Dimension::Id;
+
+    m_dimTypes = m_remoteLayout->dimTypes();
+    for (DimType& dt : m_dimTypes)
+    {
+        // If the data is stored as Laszip, then PDAL's LasReader will unscale
+        // XYZ for us.
+        if (m_info->dataType() == EptInfo::DataType::Laszip &&
+                (dt.m_id == D::X || dt.m_id == D::Y || dt.m_id == D::Z))
```

Comment:
Then wouldn't it be an error to define a scale/offset for XYZ for LASzip source?  Seems some sort of a check/notice is in order.

Reply:
The scale/offset are still specified if laszip is used as EPT storage, so this would not be an error.  Having laszip storage without scaled XYZ is invalid - that should be added to the spec but is already implicitly enforced via the nature of laszip.

Reply:
Maybe I wasn't clear.  Couldn't you have scale/offset IN the laszip file and have that be different from what's specified in the schema?  That seems problematic unless you enforce it somehow.